### PR TITLE
Fix minor typos in sharing_components.rst

### PR DESCRIPTION
**Pull request by ieshreya**
## What does this PR do?

Fix minor typos in [sharing_components.rst](https://github.com/Lightning-AI/lightning/blob/master/docs/source-app/glossary/sharing_components.rst).

### Does your PR introduce any breaking changes? If yes, please list them.

None
## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source-app/glossary/sharing_components.rst [hunk 1]
```diff
@@ -9,41 +9,42 @@ Sharing my components
 ----
 
 *********************************************
-Why should I consider sharing my components ?
+Why should I consider sharing my components?
 *********************************************
```
Comment (awaelchli) [inline docs/source-app/glossary/sharing_components.rst:13]: The underline length needs to be adjusted as well
```suggestion
Why should I consider sharing my components?
********************************************
```

File: docs/source-app/glossary/sharing_components.rst [hunk 2]
```diff
@@ -9,41 +9,42 @@ Sharing my components
...
 
-By creating components and sharing them with everyone else, the barrier to entry will be go down.
+By creating components and sharing them with everyone else, the barrier to entry will go down.
 
 ----
 
 *************************************
-How should I organize my components ?
+How should I organize my components?
 *************************************
```
Comment (awaelchli) [inline docs/source-app/glossary/sharing_components.rst:23]:
```suggestion
How should I organize my components?
************************************
```

File: docs/source-app/glossary/sharing_components.rst [hunk 3]
```diff
@@ -9,41 +9,42 @@ Sharing my components
...
 
-.. note:: As a Lightning user, it helps to implement your components thinking someone else is going use them.
+.. Note:: As a Lightning user, it helps to implement your components thinking someone else is going to use them.
 
 ----
 
 ******************************************
-How should I proceed to share components ?
+How should I proceed to share components?
 ******************************************
```
Comment (awaelchli) [inline docs/source-app/glossary/sharing_components.rst:46]:
```suggestion
How should I proceed to share components?
*****************************************
```

File: docs/source-app/glossary/sharing_components.rst [hunk 4]
```diff
@@ -9,41 +9,42 @@ Sharing my components
...
+* This also helps create an ecosystem with **reusable** components.
+
 
 Now, imagine you have implemented a **KerasScriptRunner** component for training any `Keras <https://github.com/keras-team/keras>`_ model with `Tensorboard UI <https://github.com/tensorflow/tensorboard>`_ integrated.
 
 Here are the best practices steps before sharing the component:
 
-* **Testing**: Ensure your component is well tested by following the :ref:`testing` guide.
-* **Documented**: Ensure your component has docstring and comes with some usage explications.
+* **Testing**: Ensure your component is well tested by following the ``:ref:`testing``` guide.
```
Comment (awaelchli) [inline docs/source-app/glossary/sharing_components.rst:37]: This isn't code, it is referencing a page testing.rst

Comment (awaelchli) [reply docs/source-app/glossary/sharing_components.rst:37]: The file is located in the parent directory:

```suggestion
* **Testing**: Ensure your component is well tested by following the ref:`../testing` guide.
```

**Comment by awaelchli** [comment]
Thank you ❤️

**Comment by ieshreya** [comment]
I misunderstood the `.rst` file to be `.md` for referencing pages. 
Thank you for resolving the issues! :)

**Review by Borda — APPROVED** [review]

**Pull request merged**